transform/flatten_mangler.go

```diff
@@ -112,29 +114,30 @@ func (f *FlattenMangler) flattenStruct(fieldPrefix, tagPrefix []string, sf refle
 
 		// get the underlying type after removing pointer for each member
 		// of the struct. Ignoring type
-		nestedK, _ := getUnderlyingKindType(nestedsf.Type)
+		nestedK, nestedT := getUnderlyingKindType(nestedsf.Type)
 		switch nestedK {
 		case reflect.Struct:
-			// only flatten if it doesn't implement TextUnmarshaler
-			if !nestedsf.Type.Implements(textMValue) {
-				flattened, err := f.flattenStruct(flattenedNames, flattenedTags, nestedsf)
-				if err != nil {
-					return out, err
-				}
-				out = append(out, flattened...)
-				continue
+			// don't flatten if it implements TextUnmarshaler
+			if nestedsf.Type.Implements(TextMReflectType) || nestedT.Implements(TextMReflectType) {
```

Comment:
While it's definitely more efficient to lean on pointerification here, I think I'd rather work backward in this check (and the others like it) by checking `reflect.PtrTo(nestedT).Implements` in addition to `nestedT.Implements` so it's clear that we're checking both. (it should be fine either way, since these are specifically struct-types, but I'd like to err on the side of clarity)

Reply:
Fixed